client/client.go

```diff
@@ -57,6 +61,19 @@ func CanPerformStartTLS(address string, config *Config) (connected bool, certifi
 	return true, certificate, nil
 }
 
+// CanPerformSSL checks whether a connection can be established to an address using the SSL protocol
+func CanPerformSSL(address string, config *Config) (connected bool, certificate *x509.Certificate, err error) {
+	conn, err := tls.DialWithDialer(&net.Dialer{Timeout: config.Timeout}, "tcp", address, nil)
+	if err != nil {
+	    return
+	}
+	defer conn.Close()
+
+	chain := conn.ConnectionState().VerifiedChains[0] // VerifiedChains[0] == PeerCertificates[0]
+	certificate = chain[0]
```

Comment:
Would it be possible to check the length before calling these?

I'd rather avoid risking a `panic`.

Reply:
Sure. I've added both checks in c55d680.